indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -397,12 +406,20 @@
     (errors/throw-service-error
       :bad-request
       (format "Concept-id %s and revision-id %s cannot be null" concept-id revision-id)))
-
   (let [{:keys [all-revisions-index?]} options
         concept-type (cs/concept-id->type concept-id)]
     (when (indexing-applicable? concept-type all-revisions-index?)
       (let [concept (meta-db/get-concept context concept-id revision-id)
-            parsed-concept (cp/parse-concept context concept)]
+            parsed-concept (cp/parse-concept context concept)
+            elastic-version (get-elastic-version context concept)]
+        (when (and (> revision-id 1)
```

Comment:
All of the logic looking at previous revisions to determine what to do should be in metadata-db.

I think in the `cmr.metadata-db.services.concept-service` namespace. We should already be looking at the previous revision at the time. Metadata-db should be placing a new type of message on the queue - I think we said something like :tombstone-removed?

Then in indexer we handle that message to call:
```
              (es/delete-document context [deleted-granule-index-name] deleted-granule-type-name
                                  (str "D" concept-id) revision-id elastic-version
                                  {:refresh true}))))
```